controllers/space/space_controller_test.go

```diff
@@ -1794,6 +1794,219 @@ func TestRetargetSpace(t *testing.T) {
+			)
+
+			hostClient := test.NewFakeClient(t, parentSpace, sb1, sb2, basicTier, subSpace)
+			member1Client := test.NewFakeClient(t, parentNStmplSet, subNStmplSet)
+			member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+
+			ctrl := newReconciler(hostClient, member1)
+			InitializeCounters(t,
+				NewToolchainStatus(
+					WithMetric(toolchainv1alpha1.UserSignupsPerActivationAndDomainMetricKey, toolchainv1alpha1.Metric{
+						"1,internal": 1,
+					}),
+					WithMetric(toolchainv1alpha1.MasterUserRecordsPerDomainMetricKey, toolchainv1alpha1.Metric{
+						string(metrics.Internal): 1,
+					}),
+					WithMember("member-1", WithSpaceCount(2)),
+				))
+
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(subSpace))
+
+			// then
+			require.NoError(t, err)
+			// subNSTemplateSet should have same usernames and roles as parentNSTemplateSet
+			nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, subNStmplSet.Name, member1Client).
+				HasSpaceRoles(
+					nstemplatetsettest.SpaceRole("basic-admin-123456new", "parentSpaceAdmin"),
+					nstemplatetsettest.SpaceRole("basic-viewer-123456new", "parentSpaceViewer"),
+				).
+				HasConditions(nstemplatetsettest.Provisioned())
+			AssertThatCountersAndMetrics(t).
+				HaveSpacesForCluster("member-1", 2) // space counter is unchanged
+		})
+
+		t.Run("create SpaceBindings for both parentSpace and subSpace, and expect subSpace will have merged roles and usernames", func(t *testing.T) {
+			// given a Space and its NSTemplateSet resource...
+			parentSpace := spacetest.NewSpace("parentSpace",
+				spacetest.WithTierName(basicTier.Name),
+				spacetest.WithSpecTargetCluster("member-1"),
+				spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+				spacetest.WithFinalizer())
+			parentNStmplSet := nstemplatetsettest.NewNSTemplateSet(parentSpace.Name,
+				nstemplatetsettest.WithReferencesFor(basicTier,
+					// include pre-existing users with role...
+					nstemplatetsettest.WithSpaceRole("admin", "parentAdmin"),
+				),
+				nstemplatetsettest.WithReadyCondition(),
+			)
+			// ...and their corresponding space bindings
+			sb1 := spacebindingtest.NewSpaceBinding("parentSpaceAdmin", parentSpace.Name, "admin", "signupAdmin")
+
+			// ...now create a subSpace
+			subSpace := spacetest.NewSpace("subSpace",
+				spacetest.WithTierName(basicTier.Name),
+				spacetest.WithSpecTargetCluster("member-1"),     // already provisioned on a target cluster
+				spacetest.WithStatusTargetCluster("member-1"),   // already provisioned on a target cluster
+				spacetest.WithSpecParentSpace(parentSpace.Name), // set the parentSpace spec
+				spacetest.WithFinalizer())
+			subNStmplSet := nstemplatetsettest.NewNSTemplateSet(subSpace.Name,
+				nstemplatetsettest.WithReadyCondition(),
+				nstemplatetsettest.WithReferencesFor(basicTier,
+					// add a new user with role...
```

Comment:
```suggestion
					// add an already existing user with a different name which is supposed to be dropped
```

Reply:
fixed in: [6b9e585](https://github.com/codeready-toolchain/host-operator/pull/733/commits/6b9e5853271c52ff603d67960dbfd63ba4f10904)